conans/client/graph/build_mode.py

```diff
@@ -59,8 +59,10 @@ def forced(self, conan_file, ref, with_deps_to_build=False):
 
         # Patterns to match, if package matches pattern, build is forced
         for pattern in self.patterns:
-            is_matching_name = fnmatch.fnmatchcase(ref.name, pattern)
-            is_matching_ref = fnmatch.fnmatchcase(repr(ref.copy_clear_rev()), pattern)
+            # Remove the @ at the end, to match for "conan install pkg/0.1@ --build=pkg/0.1@"
+            clean_pattern = pattern[:-1] if pattern.endswith("@") else pattern
```

Comment:
Better to clean the patterns when they are added to the `self.patterns` list

Reply:
Done. I have also added the possibility to match a complete reference including RREV, with and without user/channel